### PR TITLE
#17 support encrypt user and password in each namespace

### PR DESCRIPTION
**Pull request by cx3ptr**
1.support encrypt user and password in each namespace
2.when add or modify config of namespace, its encrypted default
3.when your config is plaintext, will read it directly and encyrpt it after next modify

### CODE REVIEW

File: proxy/server/manager.go [hunk 1]
```diff
@@ -40,7 +40,7 @@ func LoadAndCreateManager(cfg *models.Proxy) (*Manager, error) {
 	if cfg.ConfigType == models.ConfigFile {
 		root = cfg.FileConfigPath
 	}
-	namespaceConfigs, err := loadAllNamespace(cfg.ConfigType, cfg.CoordinatorAddr, cfg.UserName, cfg.Password, root)
+	namespaceConfigs, err := loadAllNamespace(cfg.ConfigType, cfg.CoordinatorAddr, cfg.UserName, cfg.Password, root, cfg.EncryptKey)
```
Comment (cx3ptr) [inline proxy/server/manager.go]: how about cfg as the param of loadAllNamespace @eastfisher #18

File: proxy/server/manager.go [hunk 2]
```diff
@@ -129,6 +129,7 @@ type Manager struct {
 	namespaces  [2]*NamespaceManager
 	users       [2]*UserManager
 	statistics  *StatisticManager
+	EncryptKey  string
```
Comment (eastfisher) [inline proxy/server/manager.go]: Is it better to move EncryptKey to struct Server?

File: models/namespace.go [hunk 3]
```diff
@@ -126,6 +129,66 @@ func (p *Namespace) Verify() error {
...
+		}
+		p.Slices[i].Password, err = encrypt(key, p.Slices[i].Password)
+		if err != nil {
+			return
+		}
+	}
+
+	return nil
+}
+
```
Comment (eastfisher) [inline models/namespace.go]: please add unit tests for Encrypt() and Decrypt()

File: proxy/server/manager.go (hunk 1 above)
Comment (eastfisher) [reply proxy/server/manager.go]: I think loadAllNamespace() needs refactoring

File: proxy/server/manager.go (hunk 2 above)
Comment (cx3ptr) [reply proxy/server/manager.go]: Yeah,i'm afraid we will use EncryptKey in manager, but it's ok now, i will change it.

File: models/namespace.go (hunk 3 above)
Comment (cx3ptr) [reply models/namespace.go]: ok，i forget it

File: proxy/server/manager.go (hunk 1 above)
Comment (cx3ptr) [reply proxy/server/manager.go]: I change params to cfg,we can refactor some code in another issue.

File: models/namespace.go (hunk 3 above)
Comment (cx3ptr) [reply models/namespace.go]: done

**Pull request merged**